guidelines/configuration-api-design.md

```diff
@@ -0,0 +1,36 @@
+## Designing configuration API's
+
+All configuration API's should be code first since that allows us to best guide the users and allows us to evolve the API's using obsolete messages.
```

Comment:
since those guidelines also apply to external contributors, should we avoid using "us"?